sql/core/src/main/scala/org/apache/spark/sql/Column.scala

```diff
@@ -1040,17 +1034,11 @@ class Column(val expr: Expression) extends Logging {
    *   df.select($"colA".name("colB"))
    * }}}
    *
-   * If the current column has metadata associated with it, this metadata will be propagated
-   * to the new column.  If this not desired, use `as` with explicitly empty metadata.
```

Comment:
Why removing these comments?

Reply:
These comments added together with the changes we just reverted. But it's good to have clear comments, I'll rephrase and add them back in the follow-up.